annotation/tests/test_annotations.py

```diff
@@ -1,85 +1,252 @@
+        user=annotator.user_id,
+        pipeline=None,
+        date=datetime(2024, 1, 1, 10, 10, 0),
+        file_id=annotation_file.file_id,
+        job_id=annotation_job.job_id,
+        pages={},
+        validated=[1],
+        failed_validation_pages=[],
+        tenant=TEST_TENANT,
+        categories=["foo", "bar"],
+        links_json=[],
+    )
+
+
+@pytest.fixture()
+def annotation_manifest():
+    yield {
+        "pages": {},
+        "validated": [1],
+        "failed_validation_pages": [],
+        "file": "path/to/file",
+        "bucket": "file-bucket",
+        "categories": [
+            {"type": "taxonomy", "value": "foo"},
+            {"type": "taxonomy", "value": "bar"},
+        ],
+    }
+
+
+def test_row_to_dict_table(annotated_doc: AnnotatedDoc):
+    result = row_to_dict(annotated_doc)
+    expected_result = {
+        "revision": "20fe52cce6a632c6eb09fdc5b3e1594f926eea69",
+        "file_id": 1,
+        "job_id": 1,
+        "user": "6ffab2dd-3605-46d4-98a1-2d20011e132d",
+        "pipeline": None,
+        "date": "2024-01-01T10:10:00",
+        "pages": {},
+        "failed_validation_pages": [],
+        "validated": [1],
+        "tenant": "test",
+        "task_id": None,
+        "categories": ["foo", "bar"],
+        "links_json": [],
+    }
+    assert result == expected_result
+
+
+def test_row_to_dict_non_table():
+    mock_dict = Mock()
+    mock_dict.__dict__ = {
+        "uuid_attr": uuid.UUID("34c665fd-ddfb-412c-a3f8-3351d87c6030"),
+        "datetime_attr": datetime(2024, 1, 1, 10, 10, 0),
+        "str_attr": "test string",
+        "int_attr": 1,
+    }
+    expected_result = {
+        "uuid_attr": "34c665fd-ddfb-412c-a3f8-3351d87c6030",
+        "datetime_attr": "2024-01-01T10:10:00",
+        "str_attr": "test string",
+        "int_attr": 1,
+    }
+    result = row_to_dict(mock_dict)
+    assert result == expected_result
 
 
-@pytest.mark.unittest
 @pytest.mark.parametrize(
-    [
-        "row",
-        "expected_dictionary",
-    ],
-    [
+    (
+        "s3_prefix",
+        "bucket_name",
+        "expected_string",
+    ),
+    (
         (
-            ANNOTATION_ROW_TABLE,
-            {
-                "uuid_attr": "34c665fd-ddfb-412c-a3f8-3351d87c6030",
-                "datetime_attr": "2024-01-01T10:10:00",
-                "str_attr": "test string",
-                "int_attr": 1,
-            },
-            # row_to_dict won't cast INTEGER to string
+            "S3_test",
+            "bucket_test",
+            "S3_test-bucket_test",
         ),
         (
-            ANNOTATION_ROW,
-            {
-                "uuid_attr": "34c665fd-ddfb-412c-a3f8-3351d87c6030",
-                "datetime_attr": "2024-01-01T10:10:00",
-                "str_attr": "test string",
-                "int_attr": 1,
-            },
-            # it return the same as the previous case
+            None,
+            "bucket_test",
+            "bucket_test",
```

Comment:
can we fit them into one line? if they are less than 79 chars ofc
like:
    `("s3_prefix", "bucket_name", "expected_string")`